### DISCUSSION TITLE
How to connect multiple files in git repository with Weblate automatically?

### DISCUSSION THREAD

**Discussion by cukrmi** [WeblateOrg/weblate#6197] [Q&A]
Hello,

Our software has two "layers" of texts that need to be localized:

1. one JSON file for all texts within the interface which is located in our infrastructure as follows 

> app/locale/LANG.json

, e.g. app/locale/cs.json (this was quite easy to connect with Weblate >> 1 project with 1 component and then required languages)
2. many short HTML files which contain texts of tooltips (occur only when you mouse hover over it). These are stored in our infrastructure as follows: 

> app/texts/LANG/a.html

, in comparison with JSON files which is only 1 per language, we have currently 200+ HTML files.

I don't know how to connect those 200+ files with Weblate.
First, I tried creating 1 component for each HTML file, but this is very time-consuming and uncomfortable to do for 200+ files. When I add new component and scan our repository, there is also found the File format: _app/texts/en/*.html_ - this is what we need to have, however, I am not able to save such a component, of course, because _Monolingual base language file_  cannot be more files thus _app/texts/en/*.html_
I had a look at the Component discovery addon, but I am not sure if it is possible to do it with this addon - I was not able to find any files - perhaps incorrect use? (please find the attached screenshot with settings within this addon)
![component-discovery](https://user-images.githubusercontent.com/20565830/122419362-226e7480-cf8b-11eb-9482-d67c26658018.png)

 Is there any option for how to do it automatically?

Thank you.

Michal

**Comment by nijel** [comment]
Seems like [component discovery](https://docs.weblate.org/en/latest/admin/addons.html#component-discovery) would help in this.
